src/types/utility.ts

```diff
@@ -0,0 +1 @@
+export interface NestedArray<T> extends Array<NestedArray<T> | T> {}
```

Comment:
I like the idea of moving this to a separate file because it's not necessarily related to jsx. Given that the helpers.ts file has a function that operates on nested arrays, could we move it there instead?